internal/validator/handler.go

```diff
@@ -44,22 +46,51 @@ type handler struct {
 	errors   chan error
 }
 
-func (this *handler) onMessage(ctx context.Context, msg *kafka.Message) {
-	request := &messageModel.IngressValidationRequest{}
-	err := json.Unmarshal(msg.Value, request)
+func (this *handler) onMessage(ctx context.Context, msgCh chan *kafka.Message) {
+	vCh := make(chan messageInfo, cap(msgCh))
 
-	if err != nil {
-		instrumentation.UnmarshallingError(ctx, err)
-		return
-	}
+	go this.validationProcess(ctx, vCh)
+
+	for {
+		if len(msgCh) <= 0 || len(vCh) >= cap(vCh){
```

Comment:
is this "busy waiting" until a message is available in the channel?

Reply:
Using `select` to block when channel is empty